### PR TITLE
better error pages

### PR DESCRIPTION
**Pull request by stevejpurves**
There are now 3 error page components

 - `ErrorSiteExpired`
 - `ErrorSiteNotFound`
 - `ErrorArticleNotFound`

And we are looking for the following behaviour when running `npm run dev`

- [x] `localhost:5001/abc/abc` results in ErrorArticleNotFound in the page area, with a TOC rendered
- [x] `localhost:5001/demo/abc` results in ErrorArticleNotFound in the page area, with a TOC rendered
- [x] `localhost:5001/abc` results in ErrorArticleNotFound in the main area (no TOC rendered)

We want to simulate the error condition in prod where the site is not deployed, or has been expired.

Looking at https://github.com/curvenote/curvespace-deploy/blob/67c47f5bf96392786b8c8e6581ef89c0b6de4e76/src/loader.server.ts#L51 we see that in prod we do a request to get the config, which returns `undefined` when dns lookup fails. 

So in local dev we can set `getConfig` to return `undefined` to simulate the following cases:

- [x] no config & `localhost:5001/abc/abc` results in ErrorSiteNotFound being rendered in the main area
- [x] no config & `localhost:5001/abc` results in ErrorSiteNotFound being rendered in the main area
- [x] no config & `localhost:5001/` results in ErrorSiteNotFound being rendered in the main area

We can further simulate an expired launchpad site by changing the check in the root `CatchBoundary` from `.startsWith('launchpad-'` to `startsWith('localhost')`, for the following cases.

- [x] no config & `localhost:5001/abc/abc` results in ErrorSiteExpired being rendered in the main area
- [x] no config & `localhost:5001/abc` results in ErrorSiteExpired being rendered in the main area
- [x] no config & `localhost:5001/` results in ErrorSiteExpired being rendered in the main area

---

## Problem

The cases tikes above are working for this PR, however, I cannot get the remaining 2 cases to work. I expect these to be picked up by the Root Catch Boundary but they are not.

Am I missing something, and the current approach just needs to be tweaked?

Or is this approach just wrong? or is there a different way to solve it, maybe not using boundaries for "no config" cases.

### CODE REVIEW

File: app/utils/loader.server.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ import { PageLoader as Data, SiteManifest } from './types';
 const contentFolder = path.join(__dirname, '..', 'app', 'content');
 
 export async function getConfig(request: Request): Promise<SiteManifest | undefined> {
-  return config;
+  return config as any;
+  // return undefined; // simulate expired or non existing deployed site
```
Comment (stevejpurves) [inline app/utils/loader.server.ts:11]: returning `undefined` here to simulate an expired site

**Comment by stevejpurves** [comment]
maybe we just shouldn't expect the root CatchBoundary to cover errors in the root loader? only its child routes..?

**Comment by stevejpurves** [comment]
we could for example maybe`return` an error response rather than throwing in the root loader when config is `undefined` and just render our error pages in place of the main site.

**Comment by rowanc1** [comment]
Added a react default export to the index page, this allowed the outlet to propagate, without it it didn't work.

File: app/routes/index.tsx [hunk 2]
```diff
@@ -1,32 +1,17 @@
...
-//           ways that the audience can learn from the simulation.
-//         </Card>
-//       </div>
-//     </main>
-//   );
-// }
+// Note this is necessary to propagate catch boundaries, even though there is a redirect
+export default function Index() {
+  return <Outlet />;
+}
```
Comment (rowanc1) [inline app/routes/index.tsx:17]: This was the fix @stevejpurves

**Pull request merged**